packages/db/src/repositories/leaderboard.ts

```diff
@@ -139,19 +139,29 @@ export class LeaderboardRepository {
     );
 
     try {
-      // Query 1: Agent basic info + global scores
+      // Query 1: Agent ranks by competition type with rank calculated in SQL
+      // Use a subquery to calculate ranks across ALL agents, then filter to requested agents
+      const rankedAgentsSubquery = this.#dbRead
```

Comment:
here's how we get the `ranks` value used in the UI and whatnot. instead of returning _just_ global scores, we also include the comp type _and_ calculate the rank in the actual query. 

(before, we'd do some post-query processing to determine the rank.)